### ISSUE TITLE
How Configure Params with Environment Variables aka HUGO_PARAMS_

### ISSUE THREAD

**Issue by gustavorps** [gohugoio/hugo#6276]
I'm trying to configure using environments variable and [in the docs says](https://gohugo.io/getting-started/configuration/#environmental-variables) to set config params, you can use the prefix the name with `HUGO_PARAMS_` so...

## This works

`config.yml`
```yml
title: Original Title
```

shell
```sh
$ env HUGO_TITLE="Some Title" npx hugo server --theme hugo-theme-material-amp
```

index.html
```html
{{ $.Site.Title }}
```

result
```html
Some Title
```

## This don't

`config.yml`
```yml
params:
  title: Original Title
```
 shell
```sh
$ env HUGO_PARAMS_TITLE="Some Title" npx hugo server --theme hugo-theme-material-amp
```

index.html
```html
{{ .Site.Params.title }}
{{ .Site.Params.Title }}
{{ .Site.Params.TITLE }}
```

result
```html
Original Title
```

**Comment by gustavorps** [comment]
:up:

**Comment by ilcardella** [comment]
I have the same problem 

config.toml
```
[params]
  email=example@mail.com
```

I use a docker-compose file where I set the environment variable
```
HUGO_PARAMS_EMAIL=my@email.com
```

The parameter is not overridden as expected

**Comment by davidkarlsson0** [comment]
Hi. It seems to work for me, using Hugo 59.0 and: 

config.toml
```
[Params]
    BranchName = "dummy"
```

shell
`$ env HUGO_PARAMS_BranchName="mr3636" hugo server`

footer.html
`.../tree/{{ .Site.Params.BranchName }}/content/...`

Result
`.../tree/mr3636/content/...`

**Comment by gustavorps** [comment]
@davidcoallier is the version, I use [hugo-bin](https://www.npmjs.com/package/hugo-bin) to reduce the complexity of my deploy but I didn't pay attention to the version issue

**Comment by Chrede88** [comment]
Hi guys,

I know this issue has been closed for quite some time now, but it the only issue I can find that resembles the issue I'm having.
I'm also trying to inject an environment variable on compile and it works for parameters with "simple" names. E.g. `env HUGO_PARAMS_MAP_zoom="15" hugo --minify`. Here I'm injecting the variable "zoom" into the params.toml file under the [map] section.

What I can't get to work, is when the parameter name has an underscore in the middle. E.g. `env HUGO_PARAMS_MAP_api_key="<my_api_key>" hugo --minify`. My guess is that the underscore is treated like all the other underscores and therefore not hitting the correct parameter. I've tried to resolve this by wrapping `api_key` in brackets, single and double quotes. I've tried escaping the underscore `api\_key`. Nothing has worked so far. What do I need to do to make this work. Any ideas?
